### DISCUSSION TITLE
String index out of range: -1

### DISCUSSION THREAD

**Discussion by ImNeverWrong** [itzg/docker-minecraft-server#1813] [Q&A]
Hi, I am running Docker with a few services and the MC-Server. The server starts up properly inside of docker and the port forwarding is setup as well. But for some reason when I try to connect to the server I always receive the error message "String index out of range: -1".
![grafik](https://user-images.githubusercontent.com/117092390/199021972-f3ec6a1e-fe86-4a92-8e26-31ed03bc1e39.png)
When I try to open the port in a brower I receive this:
![grafik](https://user-images.githubusercontent.com/117092390/199022113-333b3c41-c958-4e36-9598-df26706eaca0.png)

What might be the issue? Cause I can't seem to find something wrong.

**Comment by itzg** [comment]
You can't open a browser on the Minecraft server port since [it's a binary protocol](https://wiki.vg/Protocol). Are you putting a web proxy in front of it?

**Reply by ImNeverWrong** [reply]
Thanks for the link. I knew I wasn't really gonna get anything out of opening it in a browser, but was curious about the error message I'd receive. I'm not running it behind a web proxy. Docker is running on a seperate machine in the local network and I'm trying to connect to the server within the local network over "(IP-Address of server):25565". The port is, as far as I can tell, setup properly on Docker.

**Reply by itzg** [reply]
You seem to have realized it, but this is not a Docker problem. Either the client or server instance (you didn't say if it was vanilla, Forge, etc) is at fault.

**Reply by ImNeverWrong** [reply]
Yea I assumed it wouldn't be a docker problem, that's why I opened it as a discussion instead of a bug report. It's vanilla and I don't necessarily think it's the server since other services are running well and are accessible.
I'll check out if I can access the server from another client.

**Reply by ImNeverWrong** [reply]
I just checked it on a different client and received the same error message. I don't really understand what might be the issue here. 😅

**Comment by ImNeverWrong** [comment]
A few updates.. 
I still don't know why I can't access the server in my local network, but I forwarded the port on my router and I'm able to access the server over the public IP now, which is fine-ish.. I will have to edit the IP every time it changes, but that's acceptable. 
(I would have had to forward the port anyways, since I wanted to make the server available for my friends).

**Reply by itzg** [reply]
Oh funny, it actually works better  out and back through the router.

**Comment by caemputer** [comment]
Might not really be relevant anymore but just in case someone else has the same problem in the future and finds this, it is quite highly ranked on search engines when you look it up. It seems to be some issue with the client because the same error message is shown when the server is not running. The issue can be mitigated by creating an entry in your hosts file, for example `192.168.2.69 	homeserver` and connecting to `homeserver` instead of the ip address. Only potentially related bug I found is this: https://bugs.mojang.com/browse/MC-252402 with the same error logs.
